provisioning/utils.go

```diff
@@ -48,21 +49,30 @@ func getPodHostIP(podClient coreclientv1.PodsGetter, targetNamespace string) (st
 	return hostIP, err
 }
 
-func GetServerInternalIP(osclient osclientset.Interface) (string, error) {
+func GetServerInternalIP(osclient osclientset.Interface, config *metal3iov1alpha1.ProvisioningSpec, client kubernetes.Interface, targetNamespace string) (string, error) {
 	infra, err := osclient.ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
 	if err != nil {
 		err = fmt.Errorf("Cannot get the 'cluster' object from infrastructure API: %w", err)
 		return "", err
 	}
-
-	// NOTE(dtantsur): do we need to handle non-BareMetal platforms here?
-	if infra.Status.PlatformStatus.BareMetal == nil {
-		err = fmt.Errorf("Cannot detect server API VIP: not a baremetal platform")
+	switch infra.Status.PlatformStatus.Type {
+	case osconfigv1.BareMetalPlatformType, osconfigv1.OpenStackPlatformType, osconfigv1.VSpherePlatformType:
+		// FIXME(dtantsur): handle the new APIServerInternalIPs field and the dualstack case.
+		return infra.Status.PlatformStatus.BareMetal.APIServerInternalIP, nil
+	//NOTE(janders) how about osconfigv1.OvirtPlatformStatus, osconfigv1.KubevirtPlatformStatus, osconfigv1.NutanixPlatformStatus?
+	// they do have the APIServerInternalIP field but I suppose they don't support disabling provisioning network?
+	case osconfigv1.NonePlatformType:
+		//TODO(janders) do things the old way, reusing master code to get the patch up, further discussion needed
+		//(temporarily) re-added all the parameters needed by this code
```

Comment:
Yeah.. maybe we return `"", nil` from here and use this as a sign to fall-back in `GetIronicIP`?

Reply:
Although... it makes a confusing public API for this function, so maybe we want to take it private...

Reply:
Thank you for the pointers, all done.